### PR TITLE
FIX: set catalog throw NoSuchElementException

### PR DESCRIPTION
**Pull request by xuanyu66**
### What problem does this PR solve? 

spark.sql("use ${catalog_name}") will throw "java.util.NoSuchElementException: None.get"
Details see issue #2221.

### What is changed and how does it works?

check namespace is defined or not.

### Check List 

Tests 

 - Integration test

### CODE REVIEW

**Comment by xuanyu66** [comment]
@sunxiaoguang @shiyuhang0 @birdstorm PTAL

**Comment by xuanyu66** [comment]
@sunxiaoguang @shiyuhang0 @birdstorm PTAL

**Comment by xuanyu66** [comment]
@sunxiaoguang @shiyuhang0 @birdstorm PTAL

**Review by birdstorm — APPROVED** [review]
LGTM

**Review by shiyuhang0 — APPROVED** [review]

File: core/src/test/scala/com/pingcap/tispark/auth/TiAuthIntegrationSuite.scala [hunk 1]
```diff
@@ -82,6 +82,16 @@ class TiAuthIntegrationSuite extends SharedSQLContext {
     TiAuthorization.enableAuth = false
   }
 
+  test("Use catalog should success") {
+    if (catalogPluginMode) {
+      spark.sql(s"use tidb_catalog")
+      spark.sql(s"use $dbPrefix$dummyDatabase")
+    } else {
+      spark.sql(s"use spark_catalog")
+      spark.sql(s"use $dbPrefix$dummyDatabase")
```
[the remark this answers is not in the document]
Comment (xuanyu66) [reply core/src/test/scala/com/pingcap/tispark/auth/TiAuthIntegrationSuite.scala:91]:
```suggestion
      spark.sql(s"use default")
```

**Comment by xuanyu66** [comment]
/run-all-tests tikv=v5.3.0 tidb=v5.3.0 pd=v5.3.0 test-spark-catalog=false

**Review by sunxiaoguang — APPROVED** [review]
LGTM

File: core/src/test/scala/com/pingcap/tispark/auth/TiAuthIntegrationSuite.scala [hunk 2]
```diff
@@ -82,6 +82,16 @@ class TiAuthIntegrationSuite extends SharedSQLContext {
     TiAuthorization.enableAuth = false
   }
 
+  test("Use catalog should success") {
+    if (catalogPluginMode) {
+      spark.sql(s"use tidb_catalog")
+      spark.sql(s"use $dbPrefix$dummyDatabase")
+    } else {
+      spark.sql(s"use spark_catalog")
+      spark.sql(s"use default")
```
Comment (xuanyu66) [inline core/src/test/scala/com/pingcap/tispark/auth/TiAuthIntegrationSuite.scala:91]:
```suggestion
```

**Pull request merged**